docs/design/architecture.md

```diff
@@ -170,47 +397,89 @@ The following diagram illustrates the Kata Containers network hotplug workflow.
+### virtio SCSI
+
+If a block-based graph driver is configured,
+`virtio-scsi` is used to "share" the workload image (such as
+`busybox:latest`) into the container's environment inside the VM.
+
+### virtio FS
+
+In all other cases a [`virtio-fs`](https://virtio-fs.gitlab.io)
+(`VIRTIO`) mount point is used instead. The [agent](#agent) uses this
+mount point as the root filesystem for the container processes.
+
+For `virtio-fs`, the [runtime](#runtime) starts one `virtiofsd` daemon
+(that runs in the host context) for each VM created.
+
+### Devicemapper
 
-Kata Containers has the ability to hotplug and remove block devices, which makes it possible to use block devices for containers started after the VM has been launched.
+The
+[devicemapper `snapshotter`](https://github.com/containerd/containerd/tree/master/snapshots/devmapper)
+is a special case. The `snapshotter` uses dedicated block devices
+rather than formatted filesystems, and operates at the block level
+rather than the file level. This knowledge is used to directly use the
+underlying block device instead of the overlay file system for the
+container root file system. The block device maps to the top
+read-write layer for the overlay. This approach gives much better I/O
+performance compared to using `virtio-fs` to share the container file
+system.
 
-Users can check to see if the container uses the devicemapper block device as its rootfs by calling `mount(8)` within the container.  If the devicemapper block device
-is used, `/` will be mounted on `/dev/vda`.  Users can disable direct mounting of the underlying block device through the runtime configuration.
+#### Hot plug and unplug
+
+Kata Containers has the ability to hot plug add and hot plug remove
+block devices. This makes it possible to use block devices for
+containers started after the VM has been launched.
+
+Users can check to see if the container uses the `devicemapper` block
+device as its rootfs by calling `mount(8)` within the container.  If
+the `devicemapper` block device is used, the root filesystem (`/`)
+will be mounted from `/dev/vda`.  Users can disable direct mounting of
+the underlying block device through the runtime
+[configuration](#configuration).
 
 ## Kubernetes support
 
-[Kubernetes\*](https://github.com/kubernetes/kubernetes/) is a popular open source
+[Kubernetes](https://github.com/kubernetes/kubernetes/), or K8s is a popular open source
 container orchestration engine. In Kubernetes, a set of containers sharing resources
 such as networking, storage, mount, PID, etc. is called a
-[Pod](https://kubernetes.io/docs/user-guide/pods/).
+[pod](https://kubernetes.io/docs/user-guide/pods/).
+
 A node can have multiple pods, but at a minimum, a node within a Kubernetes cluster
```

Comment:
It might be worth identifying what a node could be. In the case of a VM as a node mention that to use Kata you need nesting enabled.